pkg/scale/decode.go

```diff
@@ -478,59 +478,87 @@ func (ds *decodeState) decodeBool(dstv reflect.Value) (err error) {
 
 // decodeUint will decode unsigned integer
 func (ds *decodeState) decodeUint(dstv reflect.Value) (err error) {
-	b, err := ds.ReadByte()
+	const maxUint32 = ^uint32(0)
+	const maxUint64 = ^uint64(0)
+	prefix, err := ds.ReadByte()
 	if err != nil {
-		return
+		return fmt.Errorf("reading from buffer: %w", err)
```

Comment:
nit for `Decoder`s, it's just an `io.Reader` so not necessarily a buffer 😉

```suggestion
		return fmt.Errorf("reading byte: %w", err)
```